services/bonita-business-data/bonita-business-data-impl/src/main/java/org/bonitasoft/engine/business/data/impl/BusinessDataServiceImpl.java

```diff
@@ -333,6 +334,27 @@ public BusinessDataQueryResult getJsonQueryEntities(final String entityClassName
         return new BusinessDataQueryResultImpl(buildJsonRepresentation((List<Entity>) list, businessDataURIPattern), businessDataQueryMetadata);
     }
 
+    private Query getCountQueryDefinition(Class<? extends Entity> businessDataClass, BusinessObject businessObject, Query queryDefinition) {
+        final Query countQueryDefinition = countQueryProvider.getCountQueryDefinition(businessObject, queryDefinition);
```

Comment:
counteQueryProvider should maybe be change, is it usefull now that we check that the auery is here on the  entity?